### PR TITLE
Xenial   startprivate

### LINKED ISSUE #145: Feature: Allow to start browser in private mode
It would be nice to be able to start browser in private mode.
I prefer such way of using browsers, for instance, cookies are not stored between sessions.

### PR DESCRIPTION
**Pull request by balcy**
fixes https://github.com/ubports/morph-browser/issues/145

### CODE REVIEW

**Comment by UniversalSuperBox** [comment]
This Jenkinsfile change does not work -- the server it's building on is not labeled armhf

**Comment by balcy** [comment]
ok undone the change

File: src/app/webbrowser/morph-browser.qml [hunk 1]
```diff
@@ -30,19 +30,19 @@ QtObject {
...
+            windowFactory.createObject(null, {"incognito": (incognito || settings.incognitoOnStart)}).show();
         }
         var window = allWindows[allWindows.length - 1];
         for (var i in urls) {
             window.addTab(urls[i]).load();
             window.tabsModel.currentIndex = window.tabsModel.count - 1;
         }
         if (window.tabsModel.count === 0) {
-            window.addTab(incognito ? "" : settings.homepage).load();
+            window.addTab(settings.homepage).load();
```
Comment (UniversalSuperBox) [inline src/app/webbrowser/morph-browser.qml:45]: Should the homepage be loaded even if the user wants to be in private mode? I'm not sure.

**Comment by balcy** [comment]
I was not sure, too, and asked it in the issue.
Assuming we treat morph started with the --incognito flag the same as morph started with the new "startprivate" flag set, not showing the homepage would mean using the new option prevents the possibility of a custom homepage for those users. (only Open new window -> Homepage would bring it up)
If a user sets a homepage they likely trust it IMO.
What about the following: We revert the line to the old state (--incognito flag is without homepage) but the "startprivate" mode is with homepage.
When the user activates the startprivate checkbox and has a homepage set, we ask wether to remove (set to "") or keep the homepage.
Alternatively we show the homepage neither with --incognito nor in startprivate mode.

**Comment by UniversalSuperBox** [comment]
It's a mixed bag on desktop, too. If you set Firefox to "Always use private browsing", it allows you to set a homepage after it restarts and keeps that homepage. If you start Chromium in incognito mode via the command line option (there's no other way), it always opens to the warning screen.

So, either seems fine.

**Comment by UniversalSuperBox** [comment]
I believe the current behavior is acceptable, if you have no further changes to make.

**Comment by balcy** [comment]
ok no changes left I think!

**Review by UniversalSuperBox — APPROVED** [review]

**Pull request merged**